### PR TITLE
Update matter-devices.xml for 5 clusters

### PR DESCRIPTION
**Pull request by lzgrablic02**
#### Problem
Basic Information, Bridged Device Basic Infoformation, Localization Configuration, Time Format Localization, and Unit Localization clusters needed to be updated in matter-devices.xml.

#### Change overview
Updated matter-device.xml to spec for the 5 clusters.

#### Testing
How was this tested? (at least one bullet point required)
* None

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -70,7 +70,9 @@ limitations under the License.
             <include cluster="Operational Credentials" client="false" server="true" clientLocked="true" serverLocked="true"></include>
             <include cluster="Localization Configuration" client="false" server="false" clientLocked="true" serverLocked="false"></include>
             <include cluster="Localization Time Format" client="false" server="false" clientLocked="true" serverLocked="false"></include>
-            <include cluster="Localization Unit" client="false" server="false" clientLocked="true" serverLocked="false"></include>
+            <include cluster="Localization Unit" client="false" server="false" clientLocked="true" serverLocked="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:73]: There's no such cluster... it's called "Unit Localization".

Similar for other clusters here: the names are off.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -70,7 +70,9 @@ limitations under the License.
...
+                <requireAttribute>TEMPERATURE_UNIT</requireAttribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:74]: Why is this required?  It's an optional attribute, iirc.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Thanks for catching that

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Yes, I was thinking mandatory for the feature, but that means it's optional. Fixed.

**Review by andy31415 — APPROVED** [review]
